go/engine/saltpack_sender_identify.go

```diff
@@ -86,9 +86,11 @@ func (e *SaltpackSenderIdentify) identifySender(ctx *Context) (err error) {
 	var lin bool
 	var uid keybase1.UID
 	if lin, uid, err = IsLoggedIn(e, ctx); err == nil && lin && uid.Equal(e.res.Uid) {
-		e.G().Log.Debug("| Sender is self")
-		e.res.SenderType = keybase1.SaltpackSenderType_SELF
-		return nil
+		if len(e.arg.userAssertion) == 0 {
```

Comment:
Don't we still need to return `SaltpackSenderType_SELF` in this case down below?